core/quantity/src/test/Ratio.test.ts

```diff
@@ -0,0 +1,456 @@
+
+      ];
+      await testRatioType("OneToN", testData);
+    });
+
+    it("ratioType precision test | NToOne", async () => {
+      const testData: { input: number, ratio: string, precision: number}[] = [
+        { input: 3, ratio: "3:1", precision: 0 },
+        { input: 3, ratio: "3:1", precision: 1 },
+        { input: 3, ratio: "3:1", precision: 2 },
+        { input: 3, ratio: "3:1", precision: 3 },
+      ];
+      await testRatioType("NToOne", testData);
+    });
+
+    it("ratioType precision test | valueBased", async () => {
+      const testData: { input: number, ratio: string, precision: number}[] = [
+        { input: 3, ratio: "3:1", precision: 0 },
+        { input: 3, ratio: "3:1", precision: 1 },
+        { input: 3, ratio: "3:1", precision: 2 },
+        { input: 3, ratio: "3:1", precision: 3 },
+      ];
+      await testRatioType("ValueBased", testData);
+    });
+  });
+
+  describe("ratio formatting that should throw an error", () => {
+    it("should throw an error if ratioType is not provided", async () => {
+      const ratioJson: FormatProps = {
+        type: "Ratio",
+        composite: {
+          includeZero: true,
+          units: [
+            { name: "Units.VERTICAL_PER_HORIZONTAL" }, // presentation unit
+          ],
+        },
+      };
+
+      const unitsProvider = new TestUnitsProvider();
+      const ratioFormat = new Format("Ratio");
+      try {
+        await ratioFormat.fromJSON(unitsProvider, ratioJson);
+        expect.fail("Expected error was not thrown");
+      } catch (e: any){
+        assert.strictEqual(e.message, "The Format Ratio is 'Ratio' type therefore the attribute 'ratioType' is required.");
+        assert.instanceOf(e, QuantityError);
+      }
+    });
+
+    it("should throw an error if ratioType is invalid", async () => {
+      const ratioJson: FormatProps = {
+        type: "Ratio",
+        ratioType: "someInvalidType",
+        composite: {
+          includeZero: true,
+          units: [
+            { name: "Units.VERTICAL_PER_HORIZONTAL" }, // presentation unit
+          ],
+        },
+      };
+
+      const unitsProvider = new TestUnitsProvider();
+      const ratioFormat = new Format("Ratio");
+      try {
+        await ratioFormat.fromJSON(unitsProvider, ratioJson);
+        expect.fail("Expected error was not thrown");
+      } catch (e: any){
+        assert.strictEqual(e.message, "The Format Ratio has an invalid 'ratioType' attribute.");
+        assert.instanceOf(e, QuantityError);
+      }
+    });
+
+  });
+
+  describe("RatioType Tests with special values", () => {
+    it("zero value", async () => {
+      const testData: { input: number, ratio: string }[] = [
+        { input: 0.0, ratio: "0:1" },
+      ];
+      await testRatioType("NToOne", testData);
+    });
+
+    it("large/small value", async () => {
+      const testData: { input: number, ratio: string }[] = [
+        { input: 0.00000001, ratio: "0:1" },
```

Comment:
might be nice to see the threshold where precision and number meet so we see with one less digit of precision it is 0:1 and with more more it is 0.00001:1